pkg/apis/configuration/validation/validation.go

```diff
@@ -156,6 +159,32 @@ func validateUpstreams(upstreams []v1alpha1.Upstream, fieldPath *field.Path, isP
 	return allErrs, upstreamNames
 }
 
+// ValidateNextUpstream checks the values given for passing queries to a upstream
+func validateNextUpstream(nextUpstream string, fieldPath *field.Path) field.ErrorList {
```

Comment:
this function validates a complicated directive. Because of that, I suggest having a separate unit test for that function with additional cases of valid/invalid values.

For example, consider the following input "error error".

Also, now need to create a map every time this function is invoked. You can put in a global variable like this one https://github.com/nginxinc/kubernetes-ingress/blob/e22d7974018929304473e92b40780cd237e4d935/pkg/apis/configuration/validation/validation.go#L441